system-x/services/hashicorp-vault/src/main/java/software/tnb/hashicorp/vault/service/HashicorpVault.java

```diff
@@ -0,0 +1,31 @@
+package software.tnb.hashicorp.vault.service;
+
+import software.tnb.common.client.NoClient;
+import software.tnb.common.deployment.WithDockerImage;
+import software.tnb.common.service.Service;
+import software.tnb.hashicorp.vault.account.HashicorpVaultAccount;
+import software.tnb.hashicorp.vault.validation.HashicorpVaultValidation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class HashicorpVault extends Service<HashicorpVaultAccount, NoClient, HashicorpVaultValidation> implements WithDockerImage {
```

Comment:
looking at `HashicorpVaultValidation`, the `NoClient` should be substituted with `VaultTemplate`

Reply:
depends on how you define `expose` - the client is `protected`  [here](https://github.com/tnb-software/TNB/blob/main/system-x/common/src/main/java/software/tnb/common/service/Service.java#L24) so to the end user it's not exposed